File: Lecture05/TodoListRecyclerView/app/src/main/java/me/bhavyaaggarwal/todolistrecyclerview/adapters/TodoRecyclerAdapter.java
package me.bhavyaaggarwal.todolistrecyclerview.adapters;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.ArrayList;

import me.bhavyaaggarwal.todolistrecyclerview.R;
import me.bhavyaaggarwal.todolistrecyclerview.models.Todo;

/**
 * Created by bhavyaaggarwal on 27/12/17.
 */

public class TodoRecyclerAdapter extends RecyclerView.Adapter<TodoRecyclerAdapter.TodoViewHolder> {
    private Context context;
    private ArrayList<Todo> todos;

    public TodoRecyclerAdapter(Context context) {
        this.context = context;
        this.todos = new ArrayList<>();
    }

    public void addTodo(Todo todo) {
        this.todos.add(todo);
        notifyDataSetChanged();
    }

    @Override
    public TodoViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater li = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        View itemView = li.inflate(R.layout.list_item_todo_item, parent, false);

        return new TodoViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(TodoViewHolder holder, final int position) {
        Todo todo = todos.get(position);

        holder.tvTask.setText(todo.getTask());
        holder.cbDone.setChecked(todo.isDone());

        holder.ivDelete.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                todos.remove(position);
                notifyDataSetChanged();
            }
        });
    }

    @Override
    public int getItemCount() {
        return todos.size();
    }

    static class TodoViewHolder extends RecyclerView.ViewHolder {
        TextView tvTask;
        CheckBox cbDone;
        ImageView ivDelete;

        public TodoViewHolder(View itemView) {
            super(itemView);
            tvTask = itemView.findViewById(R.id.tvTask);
            cbDone = itemView.findViewById(R.id.cbDone);
            ivDelete = itemView.findViewById(R.id.ivDelete);
        }
    }
}

File: Lecture06/TodoListKotlin/app/src/main/java/me/bhavyaaggarwal/todolistkotlin/adapters/TodoRecyclerAdapter.kt
package me.bhavyaaggarwal.todolistkotlin.adapters

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CompoundButton
import kotlinx.android.synthetic.main.list_item_todo_item.view.*
import me.bhavyaaggarwal.todolistkotlin.R
import me.bhavyaaggarwal.todolistkotlin.models.Todo

/**
 * Created by bhavyaaggarwal on 27/12/17.
 */
class TodoRecyclerAdapter(val context: Context, val todos: ArrayList<Todo> = ArrayList()) :
        RecyclerView.Adapter<TodoRecyclerAdapter.TodoViewHolder>() {

    override fun getItemCount(): Int {
        return todos.size;
    }

    fun addTodo(todo: Todo) {
        todos.add(todo)
        notifyDataSetChanged()
    }

    override fun onBindViewHolder(holder: TodoViewHolder?, position: Int) {
        holder?.bindView(todos[position]);
        holder?.itemView!!.cbDone.setOnCheckedChangeListener(CompoundButton.OnCheckedChangeListener { buttonView, isChecked ->
            todos[position].done = isChecked
        })
        holder?.itemView!!.ivDelete.setOnClickListener(View.OnClickListener {
            todos.removeAt(position)
            notifyDataSetChanged()
        })
    }

    override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): TodoViewHolder {
        val li: LayoutInflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater;
        val itemView = li.inflate(R.layout.list_item_todo_item, parent, false);
        return TodoViewHolder(itemView);
    }


    inner class TodoViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        fun bindView(todo: Todo) {
            itemView.tvTask.text = todo.task
            itemView.cbDone.isChecked = todo.done
        }
    }
}